flights/views.py

```diff
@@ -60,7 +62,49 @@ def get(self, request):
 
         serializer = serializers.BookingSerializer(booked_flights, many=True)
         return Response(
-            {
+            data={
                 "booked_flight": serializer.data
-            }
+            })
+
+
+class SearchFlight(generics.ListCreateAPIView):
+    """
+    Search flight view set.
+    """
+    serializer_class = serializers.SearchSerializer
+    queryset = models.Flight.objects.all()
+    permission_classes = (AllowAny,)
+    http_method_names = [u'post', u'put', u'patch', u'delete', u'head', u'options', u'trace']
+
+    def post(self, request):
+        from_location = request.POST.get('from_location')
+        to_location = request.POST.get('to_location')
+        departure_time = request.POST.get('departure_time')
+
+        if not from_location or not to_location or not departure_time:
+            return Response(
+                data={
+                    "error": "All data fields are required! - from_location, departure_time, to_location "
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        #  validate time format
+        try:
+            departure_time = datetime.strptime(departure_time, "%Y-%m-%dT%H:%M")
```

Comment:
Just like we discussed, the search should look at a date range and return results within that range.